src/main/scala/Table.scala

```diff
@@ -508,18 +507,70 @@ case class Table private (
   }
   warnings = warnings.concat(warningsFromColumns)
 
-  def validateRow(row: CSVRecord): WarningsAndErrors = {
+  def validateRow(row: CSVRecord): (
+      WarningsAndErrors,
+      List[Any],
+      List[(ForeignKeyWithTable, List[Any])],
```

Comment:
Can we create an object to hold all of these properties in? It can get confusing when passing back large tuples.

Reply:
ok, I have introduced a new object to hold all of these. I will need your help to give a better name to it!